src/rpc/state_api.rs

```diff
@@ -416,6 +416,54 @@ pub async fn state_miner_recoveries<DB: Blockstore + Send + Sync + 'static>(
         .map(|r| r.into())
 }
 
+pub async fn state_miner_available_balance<DB: Blockstore + Send + Sync + 'static>(
+    params: Params<'_>,
+    data: Data<RPCState<DB>>,
+) -> Result<LotusJson<TokenAmount>, JsonRpcError> {
+    let LotusJson((miner_address, ApiTipsetKey(tsk))): LotusJson<(Address, ApiTipsetKey)> =
+        params.parse()?;
+
+    let store = data.chain_store.blockstore();
+    let ts = data
+        .state_manager
+        .chain_store()
+        .load_required_tipset_or_heaviest(&tsk)?;
+    let actor = data
+        .state_manager
+        .get_actor(&miner_address, *ts.parent_state())?
+        .ok_or_else(|| anyhow::anyhow!("Miner actor not found"))?;
+    let state = miner::State::load(store, actor.code, actor.state)?;
+    let actor_balance: TokenAmount = actor.balance.clone().into();
+    let (vested, available): (TokenAmount, TokenAmount) = match &state {
+        miner::State::V13(s) => (
+            s.check_vested_funds(store, ts.epoch())?.into(),
```

Comment:
It looks like it could potentially be `macro`ized

Reply:
https://github.com/ChainSafe/forest/issues/4064